src/cache.ts

```diff
@@ -27,20 +35,83 @@ const getExpirationTimeoutInMilliseconds = (expiresIn: number, exp: number) => {
   return Math.min(expiresIn, expTime) * 1000 * 0.8;
 };
 
-export default class Cache {
+export class LocalStorageCache implements ICache {
+  public save(entry: CacheEntry): void {
+    const cacheKey = `@@auth0@@${createKey(entry)}`;
+    const expiresInTime = Math.floor(Date.now() / 1000) + entry.expires_in;
+
+    const expirySeconds =
+      Math.min(expiresInTime, entry.decodedToken.claims.exp) - 60; // take off a small leeway
+
+    const payload = {
+      body: entry,
+      expiresAt: expirySeconds
+    };
+
+    const timeout = getExpirationTimeoutInMilliseconds(
+      entry.expires_in,
+      entry.decodedToken.claims.exp
+    );
+
+    setTimeout(() => {
+      window.localStorage.removeItem(cacheKey);
+    }, timeout);
+
+    window.localStorage.setItem(cacheKey, JSON.stringify(payload));
+  }
+
+  public get(key: CacheKeyData): CacheEntry {
+    const cacheKey = `@@auth0@@${createKey(key)}`;
+    const json = window.localStorage.getItem(cacheKey);
+    let payload;
+
+    if (!json) return;
+
+    payload = JSON.parse(json);
+
+    if (!payload) return;
+
+    const nowSeconds = Math.floor(Date.now() / 1000);
+
+    if (payload.expiresAt < nowSeconds) {
+      window.localStorage.removeItem(cacheKey);
+      return;
+    }
+
+    return payload.body;
+  }
+
+  public clear() {
+    for (var i = localStorage.length - 1; i >= 0; i--) {
+      if (localStorage.key(i).startsWith('@@auth0@@')) {
+        localStorage.removeItem(localStorage.key(i));
+      }
+    }
+  }
+}
+
+export class InMemoryCache implements ICache {
   cache: CachedTokens = {};
```

Comment:
This should probably use a `Set` rather than an object as part of JS best practices.  (need to change set/get/clear too)

Reply:
I'd like to be able to reference individual items by key, so I could use a `Map` here.

I'd also like to think about this more carefully - `Map` and `Set` are not polyfilled so I'd like to investigate how we can do that and not break IE11 users. Do you see this is as blocking this PR?

Reply:
If we're supporting IE11 users just continue to use the object.